### PR TITLE
[Build Pipeline] Add potential additional NuGet restore packages

### PR DESCRIPTION
**Pull request by jashook**

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <!-- Don't use any higher level config files.
+       Our builds need to be isolated from user/machine state -->
+  <fallbackPackageFolders>
```
Comment (weshaggard) [inline NuGet.config]: You should probably also remove any other nuget.config files in the repo.

File: buildpipeline/DotNet-CoreClr-Trusted-Linux-Crossbuild.json [hunk 2]
```diff
@@ -103,7 +103,7 @@
       },
       "inputs": {
         "filename": "docker",
-        "arguments": "run --rm $(DockerCommonRunArgs) ./sync.sh",
+        "arguments": "run --rm $(DockerCommonRunArgs) ./sync.sh -p -- /p:DotNetRestoreSources=$(PB_RestoreSource)",
```
Comment (weshaggard) [inline buildpipeline/DotNet-CoreClr-Trusted-Linux-Crossbuild.json]: What is the `-p` option?

Comment (jashook) [reply buildpipeline/DotNet-CoreClr-Trusted-Linux-Crossbuild.json]: >-p         Restore all NuGet packages for the repository"

File: buildpipeline/DotNet-CoreClr-Trusted-Mac.json [hunk 3]
```diff
@@ -31,7 +31,7 @@
       },
       "inputs": {
         "filename": "$(Agent.BuildDirectory)/s/sync.sh",
-        "arguments": "",
+        "arguments": "-p -- /p:DotNetRestoreSources=$(PB_RestoreSource)",
```
Comment (weshaggard) [inline buildpipeline/DotNet-CoreClr-Trusted-Mac.json]: I suspect we are going to need this on more then just the sync command. Are you sure we don't do any restoring on other commands?

**Comment by RussKeldorph** [comment]
@dotnet-bot test this please

**Comment by jashook** [comment]
After speaking with @mmitche and @wtgodbe I believe CoreCLR does not need these changes for Product Construction.

**Pull request closed without merging**